engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/UserTaskTransformer.java

```diff
@@ -54,6 +63,41 @@ private void transformTaskDefinition(final JobWorkerProperties jobWorkerProperti
     jobWorkerProperties.setRetries(new StaticExpression("1"));
   }
 
+  private void transformAssignmentDefinition(
+      final UserTask element, final JobWorkerProperties jobWorkerProperties) {
+    final var assignmentDefinition =
+        element.getSingleExtensionElement(ZeebeAssignmentDefinition.class);
+    if (assignmentDefinition == null) {
+      return;
+    }
+    final var assignee = assignmentDefinition.getAssignee();
+    if (assignee != null && !assignee.isBlank()) {
+      jobWorkerProperties.setAssignee(expressionLanguage.parseExpression(assignee));
+    }
+    final var candidateGroups = assignmentDefinition.getCandidateGroups();
+    if (candidateGroups != null && !candidateGroups.isBlank()) {
+      final var candidateGroupsExpression = expressionLanguage.parseExpression(candidateGroups);
+      if (candidateGroupsExpression.isStatic()) {
+        // static candidateGroups must be in CSV format, but this is already checked by validator
+        jobWorkerProperties.setCandidateGroups(
+            ZeebeExpressionValidator.parseListOfCsv(candidateGroups)
+                .map(UserTaskTransformer::asListLiteralExpression)
+                .map(expressionLanguage::parseExpression)
+                .get());
+      } else {
+        jobWorkerProperties.setCandidateGroups(candidateGroupsExpression);
+      }
+    }
+  }
+
+  private static String asListLiteralExpression(final List<String> values) {
+    return String.format(
+        "=[%s]",
+        values.stream()
+            .map(value -> String.format("\"%s\"", value))
+            .collect(Collectors.joining(",")));
```

Comment:
:wrench: We could simplify the expression by using `Collectors.joining(delimiter, prefix, suffix)`.

```suggestion
    return values
        .stream()
        .map(value -> String.format("\"%s\"", value))
        .collect(Collectors.joining(",", "=[", "]")));
```